src/ecs/commands/runCommandInContainer.ts

```diff
@@ -31,6 +39,26 @@ export async function runCommandInContainer(
     let status: vscode.Disposable | undefined
 
     try {
+        const taskHasPermissions = await checkRequiredPermissions(node)
+        if (node.taskRoleArn === undefined || !taskHasPermissions) {
+            const viewDocsItem = localize('AWS.generic.viewDocs', 'View Documentation')
+            window
+                .showErrorMessage(
+                    localize(
+                        'AWS.command.ecs.runCommandInContainer.missingPermissions',
+                        'You may not have the permissions required for using execute command. Please set up a Task Role with the required permissions and retry.'
+                    ),
+                    viewDocsItem
+                )
+                .then(selection => {
+                    if (selection === viewDocsItem) {
+                        vscode.env.openExternal(vscode.Uri.parse(ecsRequiredPermissionsUrl))
+                    }
+                })
+            result = 'Cancelled'
```

Comment:
Should be 'Failed', not cancelled?

Let's think about adding a `reason` field to this metric and include something like "Insufficient Task Permissions" as the value (we'll need to update the common definitions to add the field). This can be a separate issue.

We don't really have an organized way yet for emitting these kinds of metrics so the value could be pretty much anything, though we should align with JB.

Reply:
Changed to 'Failed'